web/app/js/components/util/ApiHelpers.jsx

```diff
@@ -101,6 +108,12 @@ const ApiHelpers = (pathPrefix, defaultMetricsWindow = '1m') => {
       promise.cancel();
     });
   };
+  const apiErrorPropType =  PropTypes.shape({
```

Comment:
I'd move this outside of the `ApiHelpers` class. There's no reason we should be constructing (and throwing away) an object just to get the proptype.

Reply:
@admc this comment does not appear to be addressed. this const is still in the `ApiHelpers` class, and everywhere we access it we still construct and throw away the empty object (when we do `ApiHelpers("").apiErrorPropType`).

This doesn't have to happen as part of this branch; can be done in a follow up.

To fix this I would move the const outside of that object entirely:
```
diff --git a/web/app/js/components/util/ApiHelpers.jsx b/web/app/js/components/util/ApiHelpers.jsx
index bf2062d..0ae2cf3 100644
--- a/web/app/js/components/util/ApiHelpers.jsx
+++ b/web/app/js/components/util/ApiHelpers.jsx
@@ -43,17 +43,17 @@ const makeCancelable = (promise, onSuccess) => {
   };
 };

+export const apiErrorPropType =  PropTypes.shape({
+  status: PropTypes.number,
+  url: PropTypes.string,
+  statusText: PropTypes.string,
+  error: PropTypes.string
+});
+
 const ApiHelpers = (pathPrefix, defaultMetricsWindow = '1m') => {
   let metricsWindow = defaultMetricsWindow;
   const podsPath = `/api/pods`;

-  const apiErrorPropType =  PropTypes.shape({
-    status: PropTypes.number,
-    url: PropTypes.string,
-    statusText: PropTypes.string,
-    error: PropTypes.string
-  });
-
```

Then we'd also need to adjust how we import `apiErrorPropType` in the other files:
```
diff --git a/web/app/js/components/ErrorBanner.jsx b/web/app/js/components/ErrorBanner.jsx
index 75e5e5a..11791e5 100644
--- a/web/app/js/components/ErrorBanner.jsx
+++ b/web/app/js/components/ErrorBanner.jsx
@@ -1,5 +1,5 @@
 import _ from 'lodash';
-import ApiHelpers from './util/ApiHelpers.jsx';
+import { apiErrorPropType } from './util/ApiHelpers.jsx';
 import PropTypes from 'prop-types';
 import React from 'react';
 import { Col, Row } from 'antd';
@@ -16,7 +16,7 @@ class ErrorMessage extends React.Component {
   }

   static propTypes = {
-    message: ApiHelpers("").apiErrorPropType,
+    message: apiErrorPropType,
     onHideMessage: PropTypes.func
   }

diff --git a/web/app/js/components/ResourceList.jsx b/web/app/js/components/ResourceList.jsx
index af5f9b9..be62f7f 100644
--- a/web/app/js/components/ResourceList.jsx
+++ b/web/app/js/components/ResourceList.jsx
@@ -1,5 +1,5 @@
 import _ from 'lodash';
-import ApiHelpers from './util/ApiHelpers.jsx';
+import { apiErrorPropType } from './util/ApiHelpers.jsx';
 import ErrorBanner from './ErrorBanner.jsx';
 import { friendlyTitle } from './util/Utils.js';
 import MetricsTable from './MetricsTable.jsx';
@@ -19,7 +19,7 @@ export class ResourceListBase extends React.Component {

   static propTypes = {
     data: PropTypes.arrayOf(metricsPropType.isRequired).isRequired,
-    error:  ApiHelpers("").apiErrorPropType,
+    error:  apiErrorPropType,
     loading: PropTypes.bool.isRequired,
     resource: PropTypes.string.isRequired,
   }
diff --git a/web/app/js/components/Version.jsx b/web/app/js/components/Version.jsx
index 5e6b78f..c45197c 100644
--- a/web/app/js/components/Version.jsx
+++ b/web/app/js/components/Version.jsx
@@ -1,4 +1,4 @@
-import ApiHelpers from './util/ApiHelpers.jsx';
+import { apiErrorPropType } from './util/ApiHelpers.jsx';
 import { Link } from 'react-router-dom';
 import PropTypes from 'prop-types';
 import React from 'react';
@@ -13,7 +13,7 @@ class Version extends React.Component {
   }

   static propTypes = {
-    error: ApiHelpers("").apiErrorPropType,
+    error: apiErrorPropType,
     isLatest: PropTypes.bool.isRequired,
     latestVersion: PropTypes.string,
     productName: PropTypes.string,
```